drivers/pinmux/pinmux_mcux_lpc.c

```diff
@@ -87,38 +71,19 @@ static const struct pinmux_driver_api pinmux_mcux_driver_api = {
 	.input = pinmux_mcux_lpc_input,
 };
 
-#ifdef CONFIG_PINMUX_MCUX_LPC_PORT0
-static const struct pinmux_mcux_lpc_config pinmux_mcux_lpc_port0_config = {
-#ifdef IOPCTL
-	.base = IOPCTL,
-#else
-	.base = IOCON,
-	.clock_ip_name = kCLOCK_Iocon,
-#endif
-	.port_no = PORT0_IDX,
-};
-
-DEVICE_DEFINE(pinmux_port0, CONFIG_PINMUX_MCUX_LPC_PORT0_NAME,
-		    &pinmux_mcux_lpc_init, device_pm_control_nop,
-		    NULL, &pinmux_mcux_lpc_port0_config,
-		    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
-		    &pinmux_mcux_driver_api);
-#endif /* CONFIG_PINMUX_MCUX_LPC_PORT0 */
-
-#ifdef CONFIG_PINMUX_MCUX_LPC_PORT1
-static const struct pinmux_mcux_lpc_config pinmux_mcux_lpc_port1_config = {
-#ifdef IOPCTL
-	.base = IOPCTL,
-#else
-	.base = IOCON,
-	.clock_ip_name = kCLOCK_Iocon,
-#endif
-	.port_no = PORT1_IDX,
-};
-
-DEVICE_DEFINE(pinmux_port1, CONFIG_PINMUX_MCUX_LPC_PORT1_NAME,
-		    &pinmux_mcux_lpc_init, device_pm_control_nop,
-		    NULL, &pinmux_mcux_lpc_port1_config,
-		    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,
-		    &pinmux_mcux_driver_api);
-#endif /* CONFIG_PINMUX_MCUX_LPC_PORT1 */
+#define PINMUX_LPC_INIT(n)						\
+	static const struct pinmux_mcux_lpc_config			\
+		pinmux_mcux_lpc_port##n##_cfg = {			\
+			.base = (uint32_t *)DT_INST_REG_ADDR(n),	\
+			.clock_ip_name = kCLOCK_Iocon,			\
```

Comment:
Is this correct when we have IOPCTL?

Reply:
Changed it to be kCLOCK_IpInvalid for IOPCTL case.